### PR TITLE
Restore self signing utils

### PR DESCRIPTION
**Pull request by ekr**
This PR cleans up fallout from PR#167 and PR#170. Specifically:

1. Exposes a public MakeNewSelfSignedCert API because need that several places.
2. Modifies the server programs so that they explicitly call the new API (otherwise they don't work).
3. Adds a flag to the clients to skip cert verification.

In future, it would help if we were more careful

### CODE REVIEW

File: crypto.go [hunk 1]
```diff
@@ -616,3 +618,50 @@ func makeTrafficKeys(params CipherSuiteParams, secret []byte) keySet {
 		iv:     HkdfExpandLabel(params.Hash, secret, "iv", []byte{}, params.IvLen),
 	}
 }
+
+func MakeNewSelfSignedCert(name string, alg SignatureScheme) (crypto.Signer, *x509.Certificate, error) {
```
Comment (marten-seemann) [inline crypto.go]: I don't think generating a self-signed certificate should be part of a TLS library.

Comment (ekr) [reply crypto.go]: Yeah, I obviously don't agree with this. We have two test programs that rely on it and I also use it in Minq. So, I really don't think it's sensible to inconvenience people out of a sense of API purity.

**Comment by ekr** [comment]
On the topic of CI, there's no reason we can't write a shell script to test these programs. It just needs doing.

File: bin/mint-server-https/main.go [hunk 2]
```diff
@@ -153,6 +153,12 @@ func main() {
 		log.Fatalf("Error: %v", err)
 	}
 
+	if certFile == "" && keyFile == "" {
+		var cert *x509.Certificate
+		priv, cert, err = mint.MakeNewSelfSignedCert(serverName, mint.RSA_PKCS1_SHA256)
+		certChain = []*x509.Certificate{cert}
+
```
Comment (bifurcation) [inline bin/mint-server-https/main.go]: Nit: Extra blank line

File: bin/mint-server-https/main.go [hunk 3]
```diff
@@ -153,6 +153,12 @@ func main() {
 		log.Fatalf("Error: %v", err)
 	}
 
+	if certFile == "" && keyFile == "" {
```
Comment (bifurcation) [inline bin/mint-server-https/main.go]: Let's explicitly signal this with a flag.

Comment (bifurcation) [reply bin/mint-server-https/main.go]: ... and obviously fail if these files are missing and the flag isn't set.

File: bin/mint-server-https/main.go [hunk 4]
```diff
@@ -177,15 +183,16 @@ func main() {
 
 	config.SendSessionTickets = sendTickets
 
-	if certChain != nil && priv != nil {
+	if certFile != "" && keyFile != "" {
```
Comment (bifurcation) [inline bin/mint-server-https/main.go]: Seems like this should get consolidated with the above:
```
switch {
case selfSign: /* generate a key/cert */
case certFile != "" && keyFile != "": /* load a key/cert */
default: /* fail */
}
```

File: conn_test.go [hunk 5]
```diff
@@ -180,19 +145,11 @@ const (
...
-	serverKey, err := rsa.GenerateKey(rand.Reader, 2048)
-	if err != nil {
-		panic(err)
-	}
-	serverCert, err = newSelfSigned(serverName, RSA_PKCS1_SHA256, serverKey)
-	if err != nil {
-		panic(err)
-	}
-	clientKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	serverKey, serverCert, err = MakeNewSelfSignedCert(serverName, RSA_PKCS1_SHA256)
```
Comment (bifurcation) [inline conn_test.go]: Y U NO ECDSA

File: crypto.go (hunk 1 above)
Comment (bifurcation) [reply crypto.go]: +1 to EKR.  We've already crossed the bridge of importing PKI gunk by following the `crypto/tls` approach of having a built-in validation system.  This is a sensible utility to offer folks.

File: conn_test.go (hunk 5 above)
Comment (ekr) [reply conn_test.go]: I can change to ECDSA. The existing code had RSA, so I was just trying to avoid changing the behavior

Comment (ekr) [reply conn_test.go]: @bifurcation: just LMK

**Pull request merged**